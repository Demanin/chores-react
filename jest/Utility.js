import crypto from 'crypto';

const getRandomString = (length = 20) => {
  return crypto.randomBytes(length).toString('hex');
};

const getRandomNumber = (min = 0, max = 9999999) => {
  min = Math.ceil(min);
  max = Math.floor(max);

  return Math.floor(Math.random() * (max - min)) + min;
};

const getRandomBool = () => {
  return !!getRandomNumber(0, 1);
};

export default {
  getRandomString,
  getRandomNumber,
  getRandomBool,
};
