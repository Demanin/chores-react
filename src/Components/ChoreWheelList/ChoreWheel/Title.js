import React, { Component } from 'react';
import PropTypes from 'prop-types';
import TitleForm from 'Components/ChoreWheelList/ChoreWheel/TitleForm';

class Title extends Component
{
  render() {
    if (!this.props.choreWheel.title.isEditable) {
      return (
        <h3
          style={{marginTop: '0px'}}
          onClick={this.props.onEditClick}
        >
          {this.props.choreWheel.title.text}
        </h3>
      );
    }

    return (
      <TitleForm
        choreWheel={this.props.choreWheel}
        title={this.props.choreWheel.title.text}
        onSave={this.props.onSaveClick}
      />
    );
  }
}

Title.propTypes = {
  choreWheel: PropTypes.shape({
    id: PropTypes.number,
    turnList: PropTypes.arrayOf(
      PropTypes.shape({ userId: PropTypes.number }),
    ),
    title: PropTypes.shape({
      text: PropTypes.string,
      isEditable: PropTypes.bool,
    }),
    isVisible: PropTypes.bool,
    priority: PropTypes.number,
  }),
  onEditClick: PropTypes.func,
  onSaveClick: PropTypes.func,
};

export default Title;
