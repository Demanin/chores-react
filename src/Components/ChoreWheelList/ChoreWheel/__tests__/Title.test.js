import React from 'react';
import Title from 'src/Components/ChoreWheelList/ChoreWheel/Title';
import Utility from 'jest/Utility';
import { shallow } from 'enzyme';

describe('Title', () => {
  let onEditClick;
  let onSaveClick;

  beforeEach(() => {
    onEditClick = jest.fn();
    onSaveClick = jest.fn();
  });

  test('non-editable title snapshot', () => {
    const fixture = shallow(<Title
      choreWheel={{ title: { isEditable: false } }}
      {...{ onEditClick, onSaveClick }}
    />);

    expect(fixture).toMatchSnapshot();
  });

  test('editable title snapshot', () => {
    const fixture = shallow(<Title
      choreWheel={{ title: { isEditable: true } }}
      {...{ onEditClick, onSaveClick }}
    />);

    expect(fixture).toMatchSnapshot();
  });

  test('edit click called', () => {
    const fixture = shallow(<Title
      choreWheel={{ title: { isEditable: false } }}
      {...{ onEditClick, onSaveClick }}
    />);

    fixture.simulate('click');

    expect(onEditClick).toHaveBeenCalled();
    expect(onSaveClick).not.toHaveBeenCalled();
  });

  test('non-editable displays text in header', () => {
    const text = Utility.getRandomString();

    const fixture = shallow(<Title
      choreWheel={{ title: { isEditable: false, text } }}
      {...{ onEditClick, onSaveClick }}
    />);

    expect(fixture.text()).toEqual(text);
  });

  test('editable title props', () => {
    const id = Utility.getRandomNumber();
    const text = Utility.getRandomString();
    const choreWheel = { title: { isEditable: true, text },  id };

    const fixture = shallow(<Title
      {...{ choreWheel, onEditClick, onSaveClick }}
    />);

    const expected = {
      choreWheel,
      title: text,
      onSave: onSaveClick,
    };
    expect(fixture.props()).toEqual(expected);
  });
});
