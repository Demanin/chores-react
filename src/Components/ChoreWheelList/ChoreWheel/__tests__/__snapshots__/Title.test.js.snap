// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Title editable title snapshot 1`] = `ShallowWrapper {}`;

exports[`Title non-editable title snapshot 1`] = `ShallowWrapper {}`;
